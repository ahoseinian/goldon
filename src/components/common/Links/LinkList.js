import React from 'react';
import SingleLink from './SingleLink'

const LinkList = ({links}) => links.length
  ? (
    <div>
      {links.map(x => <SingleLink {...x} key={x._id}/>)}
    </div>
  )
  : null;

LinkList.propTypes = {
  links: React.PropTypes.arrayOf(React.PropTypes.object)
};

export default LinkList;
