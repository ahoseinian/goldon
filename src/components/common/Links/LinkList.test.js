import React from 'react'
import {shallow} from 'enzyme'
import LinkList from './LinkList'
import toJson from 'enzyme-to-json'

describe('LinkList', () => {
  it('renders links', () => {
    const links = [
      {_id:1},
      {_id:2},
      {_id:3},
    ]
    const wrapper = shallow(<LinkList links={links}/>)

    expect(
      toJson(wrapper)
    ).toMatchSnapshot()
  })
})
