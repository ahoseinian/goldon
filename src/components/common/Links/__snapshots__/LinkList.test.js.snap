exports[`LinkList renders links 1`] = `
<div>
  <SingleLink
    _id={1}
    name=""
    url="" />
  <SingleLink
    _id={2}
    name=""
    url="" />
  <SingleLink
    _id={3}
    name=""
    url="" />
</div>
`;
